File: src/main/java/land/leets/domain/application/domain/Application.java
package land.leets.domain.application.domain;

import java.time.LocalDateTime;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.OneToOne;
import land.leets.domain.application.type.ApplicationStatus;
import land.leets.domain.application.type.Position;
import land.leets.domain.application.type.SubmitStatus;
import land.leets.domain.shared.BaseTimeEntity;
import land.leets.domain.user.domain.User;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity(name = "applications")
@Builder
@Getter
@Setter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor
public class Application extends BaseTimeEntity {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;

	@OneToOne
	@JoinColumn(name = "uid")
	private User user;

	@Column(nullable = false)
	private String name;

	@Column(nullable = false)
	private String major;

	@Column(nullable = false)
	private String grade;

	@Column
	private String project;

	@Column
	private String algorithm;

	@Column
	private String portfolio;

	@Column(nullable = false, columnDefinition = "char(10)")
	@Enumerated(EnumType.STRING)
	private Position position;

	@Column
	private String career;

	@Column(nullable = false)
	private String interviewDay;

	@Column(nullable = false)
	private String interviewTime;

	@Column(nullable = false, columnDefinition = "text(600)")
	private String motive;

	@Column(nullable = false, columnDefinition = "text(600)")
	private String expectation;

	@Column(nullable = false, columnDefinition = "text(600)")
	private String capability;

	@Column(nullable = false, columnDefinition = "text(600)")
	private String conflict;

	@Column(nullable = false, columnDefinition = "text(600)")
	private String passion;

	@Column
	private LocalDateTime appliedAt;

	@Builder.Default
	@Column(columnDefinition = "char(10)")
	@Enumerated(EnumType.STRING)
	private ApplicationStatus applicationStatus = ApplicationStatus.PENDING;

	@Column(columnDefinition = "char(10)")
	@Enumerated(EnumType.STRING)
	private SubmitStatus submitStatus;

	public void updateInfo(LocalDateTime appliedAt) {
		this.appliedAt = appliedAt;
	}
}

File: src/main/kotlin/land/leets/domain/application/domain/Application.kt
package land.leets.domain.application.domain

import jakarta.persistence.*
import land.leets.domain.application.presentation.dto.ApplicationRequest
import land.leets.domain.application.type.ApplicationStatus
import land.leets.domain.application.type.Position
import land.leets.domain.application.type.SubmitStatus
import land.leets.domain.shared.BaseTimeEntity
import land.leets.domain.user.domain.User
import java.time.LocalDateTime

@Entity(name = "applications")
class Application(
    @OneToOne
    @JoinColumn(name = "user_id")
    var user: User,

    @Column(nullable = false)
    var name: String,

    @Column(nullable = false)
    var major: String,

    @Column(nullable = false)
    var grade: String,

    @Column
    var project: String? = null,

    @Column
    var algorithm: String? = null,

    @Column
    var portfolio: String? = null,

    @Column(nullable = false, columnDefinition = "char(10)")
    @Enumerated(EnumType.STRING)
    var position: Position,

    @Column
    var career: String? = null,

    @Column(nullable = false)
    var interviewDay: String,

    @Column(nullable = false)
    var interviewTime: String,

    @Column(nullable = false, columnDefinition = "text(600)")
    var motive: String,

    @Column(nullable = false, columnDefinition = "text(600)")
    var expectation: String,

    @Column(nullable = false, columnDefinition = "text(600)")
    var capability: String,

    @Column(nullable = false, columnDefinition = "text(600)")
    var conflict: String,

    @Column(nullable = false, columnDefinition = "text(600)")
    var passion: String,

    @Column
    var appliedAt: LocalDateTime? = null,

    @Column(columnDefinition = "char(10)")
    @Enumerated(EnumType.STRING)
    var applicationStatus: ApplicationStatus = ApplicationStatus.PENDING,

    @Column(columnDefinition = "char(10)")
    @Enumerated(EnumType.STRING)
    var submitStatus: SubmitStatus,

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null
) : BaseTimeEntity() {
    fun updateInfo(appliedAt: LocalDateTime) {
        this.appliedAt = appliedAt
    }

    fun updateApplicationStatus(applicationStatus: ApplicationStatus) {
        this.applicationStatus = applicationStatus
    }

    fun updateContent(request: ApplicationRequest) {
        this.name = request.name
        this.major = request.major
        this.grade = request.grade
        this.project = request.project
        this.algorithm = request.algorithm
        this.portfolio = request.portfolio
        this.position = request.position
        this.career = request.career
        this.interviewDay = request.interviewDay
        this.interviewTime = request.interviewTime
        this.motive = request.motive
        this.expectation = request.expectation
        this.capability = request.capability
        this.conflict = request.conflict
        this.passion = request.passion
        this.submitStatus = request.submitStatus
    }
}
